### ISSUE TITLE
How can I use nerdamer to...

### ISSUE THREAD

**Issue by Happypig375** [jiggzson/nerdamer#255]
1. Find the value of 27^x+27^(-x) if 3^x+3^(-x)=4
2. Find the value of 1/a-1/b if (12.5)^a=1000 and (0.0125)^b=1000
3. Find the value of x^6+x^(-6) if x+x^(-1)=3, x>0
4. Simplify sqrt(x+sqrt(x^2+1)) if x=1/2(a^2-a^(-2)) and a>0
5. Find the value of a^(2x) if (a^(3x)+a^(-3x))/(a^x+a^(-x))=1
6. Find the value of x*y*z if m=a^x, n=a^y, m^y*n^x=a^(2/z), a>0, z!=0
7. Find a relation connecting x, y and z if 2^(3x)=7^(4y)=28^(5z)
8. Find the value of 2/x+3/y in terms of z if 8^x=9^y=6^z
9. Find the value of x+y+z if a^(1/x)=b^(1/y)=c^(1/z)=a*b*c

There are currently no obvious way to calculate these using nerdamer.

**Comment by Happypig375** [comment]
My answers:
1. 52
2. 1
3. 322
4. a
5. 1
6. 1
7-9. Still calculating :stuck_out_tongue:

**Comment by Happypig375** [comment]
These set of questions are the primary motivation for #233.

**Comment by jiggzson** [comment]
@Happypig375. Do me a favor. Create an issue called "Happypig375 wishlist" and move all your issues there. Update that issue as you go. I have 79 open issues with no end in sight. Some are duplicates of other issues, some are wishlist items. If it's a bug by all means create a new issue, otherwise please refrain from creating more needless issues. Thanks.

**Comment by Happypig375** [comment]
Sorry I was not free yesterday, but was there a message about creating a wish list...?
![image](https://user-images.githubusercontent.com/19922066/30856248-5e86217e-a2ea-11e7-88be-c23673386bd7.png)

**Comment by jiggzson** [comment]
Yes. I wrote that message but it sounded a bit mean after rereading it and I wanted to phrase it better. Can you please stop creating an issue when it's clearly related to an existing issue and a simple update of that issue would suffice. Example: In #210 you commented
>#119 (my first ever issue on here pops up again) is already a big simplification. I would not see x^2+4x-45 and x^2+x-30 as having any common factors unless I factorize it specifically.

Yet you turn around and create issue #256. I got the message clearly with your original comment. In fact, you even identified how it should be solved. Another example: This issue is clearly related to solving equations with variable exponents. There was no need to create this one. You recently created two issues related to intervals and before that three issues related to units. I appreciate the feedback and I know you mean well but let's keep it somewhat organized. Creating an issue for everything does not help.

**Comment by Happypig375** [comment]
>Yet you turn around and create issue #256.

#256 is a bug with divide. It should have returned (x+9)/(x+6) but it returns (-15+3*x)*(-30+x+x^2)^(-1)+1.

**Comment by Happypig375** [comment]
>You recently created two issues related to intervals

Integer intervals are not required to implement solving inequalities so it is created separately.

**Comment by Happypig375** [comment]
>three issues related to units

#230 is for general units
#185 is for a global setting since angle units, take radians for example, is m m^-1, aka no unit.
#248: 
>Well, those do not solve the problem of replacing builtin functions with custom implementations.

#154, #255 unrelated

**Comment by Happypig375** [comment]
>Creating an issue for everything does not help.

Okay 😃

**Comment by Happypig375** [comment]
>let's keep it somewhat organized

Remember #183?

**Comment by Happypig375** [comment]
>However I do care about how the library performs

So performance is your top priority?

**Comment by Happypig375** [comment]
>The fact that you're not getting the simplest answer doesn't make it a bug.

http://nerdamer.com/functions/divide.html
>Returns the appropriate value if possible otherwise it returns the function with the **simplified** expression

Did I misunderstand this line?

**Comment by jiggzson** [comment]
> So performance is your top priority? Is "performance" speed or memory usage?

Accuracy is my top priority but since I use the symbolic layer in other functions, integrals, derivatives, etc., and I don't really know where I'm going to stop development, I'd prefer to keep the core lean. 
> Did I misunderstand this line?

I'm not disputing your claim. I simply disagree that it's a bug. My understanding of a bug is that the output is invalid.

**Comment by Happypig375** [comment]
>My understanding of a bug is that the output is invalid.

I think that a bug is that the result is incorrect (including invalid).
In this case, divide is supposed to return the function with the simplified expression.
(-15+3*x)*(-30+x+x^2)^(-1)+1
=(-15+3*x)/(-30+x+x^2)+1
=(3(x-5))/((x-5)(x+6))+1
=3/(x+6)+1

It is not (x+9)/(x+6), but either way, the result is still not in the most simplified form.

**Comment by Happypig375** [comment]
>I'd prefer to keep the core lean.

You can split some functions into another new module.

**Comment by Happypig375** [comment]
>Accuracy is my top priority

I would also like to see #196 and #221 implemented.
>I don't really know where I'm going to stop development

You are going to stop developing what?

**Comment by jiggzson** [comment]
@Happypig375, this issue belongs under #322. Would you agree? Either that or we have to pick a better name since I keep overlooking this issue.

**Comment by Happypig375** [comment]
Hmm...
What about use cases for #350?
